keps/0022-diagnostics-bundle.md

```diff
@@ -0,0 +1,389 @@
+      commands, Elasticsearch's
+      [HTTP API](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html)
+      responses, etc.)
+
+3.  [dcos/dcos-diagnostics](https://github.com/dcos/dcos-diagnostics)
+
+    Does diagnostics collection for [DC/OS](https://dcos.io/) clusters.
+
+4.  [mesosphere/bun](https://github.com/mesosphere/bun)
+
+    Does diagnostics analysis for archives created with `dcos/dcos-diagnostics`.
+
+    It is also important to notice that some applications have existing tooling
+    for application-level diagnostics collection, either built by the supporting
+    organizations behind the applications or the community. A few examples:
+
+    - [Elasticsearch's support-diagnostics](https://github.com/elastic/support-diagnostics)
+    - [Apache Kafka's System Tools](https://cwiki.apache.org/confluence/display/KAFKA/System+Tools)
+
+## Concepts
+
+### Fault
+
+One component of the system deviating from its specification.
+
+### Failure
+
+The system as a whole stops providing the required service to the user.
+
+### Operator
+
+A KUDO-based
+[Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/),
+e.g. [kudo-cassandra](https://github.com/mesosphere/kudo-cassandra-operator),
+[kudo-kafka](https://github.com/mesosphere/kudo-kafka-operator).
+
+### Operator instance
+
+An operator instance.
+
+### Application
+
+Underlying software that is managed by an operator instance, e.g., Apache
+Cassandra, Apache Kafka, Elasticsearch, etc.
+
+### Operator developer
+
+Someone who builds and maintains operators.
+
+### Operator user
+
+Someone who installs and maintains operator instances.
+
+### Diagnostic artifact
+
+A file, network response, or command output that contains information that is
+potentially helpful for operator users to diagnose faults with their operator
+instances, and for operator users to provide to operator developers and/or
+people who support operators.
+
+## Goals
+
+### Functional
+
+- Collect "Kubernetes workload"-specific diagnostics artifacts related to an
+  operator instance
+- Collect KUDO-specific diagnostics artifacts related to an operator instance
+- Collect application-specific diagnostics artifacts related to an operator
+  instance
+- Bundle all diagnostics artifacts into an archive
+
+### Non-functional
+
+1.  Provide an **easy** experience for operator users to collect diagnostic
+    artifact archives
+2.  Be resilient to faults and failures. Collect as much diagnostics artifacts
+    as possible and allow failed collections to be retried (idempotency) and
+    incremented to (like `wget`'s `--continue` flag), in a way that collection
+    is _resumable_
+3.  Incorporate standard tools that are already provided by either organizations
+    or the community behind applications as much as possible
+
+## Non-goals
+
+- Extensive collection of Kubernetes-related diagnostics artifacts
```

Comment:
But this tool *will* need to do this. I'm assuming you meant something like the following:

```suggestion
- Collection of Kubernetes-related diagnostics artifacts that are not related to diagnosing issues with KUDO operators/instances.
```